astroid/_ast.py

```diff
@@ -1,3 +1,6 @@
+# Licensed under the LGPL: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
+# For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/astroid/graphs/contributors
```

Comment:
Can we make the `pre-commit` to add a new line here?

Reply:
It's pretty complicated, we're checking that the text exists, and if it does not then we add it. If we change the copyright.txt with a newline, it would "work" but removing a line before the docstring would make the script add everything again not replace what's existing. Imo this is "good enough" ™️ but we could modify the pre-commit hook too, what do you think ?

Reply:
I'm not sure what you mean "a line before the docstring". There is not docstring here right?

Reply:
Say we have this file:
```python
# Copyright
from a import b
```
And the copyright.txt is this:
```shell
# Copyright


```
Then the result will be this:
```python
# Copyright

# Copyright
from a import b
```
Unless we change the way the pre-commit hook works this is not ideal.

Reply:
Hmm yeah, we should probably try and change the hook then. Should be fairly trivial to add a `--new-line` flag.

```python
if not token[i + 1] == token.NL:
    # add token
```

Or something 😄

Reply:
It would be trivial, if the code was working with tokens like a very good docstring linter I know 😄 But it's working using bytes directly (thus the issue with line ending) https://github.com/Pierre-Sassoulas/copyright_notice_precommit/blob/main/scripts/copyright_notice.py#L71

Reply:
Hmm, but in your `--auto-fix` PR can't you do something like:
```python
if config.new_line:
    lines_to_write += "\n"
file.write(lines_to_write)
```

Reply:
I think it's more complicated than that, you'd have to detect that some line already exists to add only a line break instead of adding everything again, you'd also have to handle the line ending for linux/windows with input copyright.txt containing line break for linux/windows (?)

Reply:
Hm, time for `pycopyrightformatter` I guess? 😄 

But I do think we shouldn't add a `pre-commit` hook if it doesn't really satisfy our needs. In its current form this will always require us to do manual input after `pre-commit` has been ran against a new file. Can we do this change without the `pre-commit` hook?

Reply:
I think it's good enough because we have no reason to change the beginning of files and once it's done it's done. I think I can do it properly now and make sure new files are done properly later.

Reply:
ie. the important thing is the copyright notice, not the line break after it